src/App.jsx

```diff
@@ -14,33 +15,86 @@
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState('');
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const handleSortAlphabetically = () => {
+    const sortedGoods = [...goods].sort();
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    setGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+    setSortType('alphabet');
+    setIsReversed(false);
+  };
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const handleSortByLength = () => {
+    const sortedGoods = [...goods].sort((a, b) => a.length - b.length);
+
+    setGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+    setSortType('length');
+    setIsReversed(false);
+  };
+
+  const handleReverse = () => {
+    setGoods([...goods].reverse());
+    setIsReversed(!isReversed);
+  };
+
+  const handleReset = () => {
+    setGoods(goods);
+    setSortType('');
+    setIsReversed(false);
+  };
```

Comment:
The method's name should start with a verb, as per the checklist. Consider renaming `handleReset` to `resetGoods`.